### PR TITLE
Resource/dhcp server

### LINKED ISSUE #60: Support provisioning DHCP Server and DHCP lease script
Setting up a DHCP server can be done following the instructions here: https://wiki.mikrotik.com/wiki/Manual:IP/DHCP_Server#Quick_Setup_Guide

Setting up DHCP lease scripts can be done using the instructions here:
https://wiki.mikrotik.com/wiki/Manual:Scripting

```
/system script
add name=myLeaseScript policy=\
    ftp,reboot,read,write,policy,test,winbox,password,sniff,sensitive,api \
    source=":log info \$leaseActIP\r\
    \n:log info \$leaseActMAC\r\
    \n:log info \$leaseServerName\r\
    \n:log info \$leaseBound"

/ip dhcp-server set  myServer lease-script=myLeaseScript
```

### PR DESCRIPTION
**Pull request by maksym-nazarenko**
Closes #60 

- [x] Resource documentation

### CODE REVIEW

File: mikrotik/resource_dhcp_server.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"interface": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "*0",
```
Comment (ddelnano) [inline mikrotik/resource_dhcp_server.go:44]: What does this default value correspond to? The example you made and what I see in my home network use friendly names and the docs aren't clear on what is 'valid'.

File: mikrotik/resource_dhcp_server.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		UpdateContext: updateDhcpServer,
+		DeleteContext: deleteDhcpServer,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"add_arp": {
+				Type:     schema.TypeBool,
+				Optional: true,
```
Comment (ddelnano) [inline mikrotik/resource_dhcp_server.go:24]: The mikrotik default for this is false. Is there a reason you decided to stick to true? I'm not sure I understand the use case of this setting from the mikrotik docs, but on my network this is disabled.

File: docs/resources/dhcp_server.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# mikrotik_dhcp_server
```
Comment (ddelnano) [inline docs/resources/dhcp_server.md:1]: You probably were already thinking about this, but please update to use `tfplugindocs`.

File: mikrotik/resource_dhcp_server.go (hunk 1 above)
Comment (maxim-nazarenko) [reply mikrotik/resource_dhcp_server.go:44]: It is not possible to create a DHCP server w/o assigning it to an interface in console, but if you use API:
```go
func TestAddDhcpServerUpdateAndDelete(t *testing.T) {
	c := NewClient(GetConfigFromEnv())

	name := "myserver"
	disabled := true
	dhcpServer, err := c.AddDhcpServer(&DhcpServer{
		Name:     name,
		Disabled: disabled,
	})
	if err != nil {
		t.Fatal(err)
	}
.....
}
```

the list of servers would be:
```shell
[admin@MikroTik] > /ip dhcp-server print
Flags: D - dynamic, X - disabled, I - invalid
 #    NAME                        INTERFACE                       RELAY           ADDRESS-POOL                       LEASE-TIME ADD-ARP
 0  X dhcp1                       *0                                              static-only                        10m
 1  X dhcp2                       *0                                              static-only                        10m
 2  X dhcp3                       *0                                              static-only                        10m
 3  X dhcp4                       *0                                              static-only                        10m
 4  X dhcp-server                 *0                                              static-only                        10m
 5  X myserver                    *0                                              static-only                        10m
[admin@MikroTik] >
```

notice the `myserver` entry.

The question could lead to necessity of `required` flag in struct tags to indicate that certain field cannot be empty:
```go
type DhcpServer struct {
	Id            string `mikrotik:".id"`
.....
	Interface     string `mikrotik:"interface,required"`
.....
}
```

Comment (ddelnano) [reply mikrotik/resource_dhcp_server.go:44]: ah, I see. Thanks for that explanation.

File: mikrotik/resource_dhcp_server.go (hunk 2 above)
Comment (ddelnano) [reply mikrotik/resource_dhcp_server.go:24]: Yep, you are correct 🤦

**Comment by ddelnano** [comment]
@maxim-nazarenko please rebase when you have the chance and we can get this merged.

**Comment by ddelnano** [comment]
Looks good, merging!

**Pull request merged**